File: src/solutions/0800-0899/868. 二进制间距.ts
function binaryGap(n: number): number {
  let max = -1;
  let lastOne = -1;
  const binary = toBinary(n);
  binary.split("").forEach((char, index) => {
    if (char === "1") {
      if (lastOne !== -1) {
        max = Math.max(max, index - lastOne);
      }
      lastOne = index;
    }
  });
  return max === -1 ? 0 : max;
}

const toBinary = (n: number): string => {
  let result = "";
  while (n > 0) {
    result = (n % 2) + result;
    n = Math.floor(n / 2);
  }
  return result;
};

File: src/solutions/0800-0899/868. 二进制间距.java
class Solution {
    public int binaryGap(int n) {
        int last = -1;
        int max = 0;
        for (int i = 0; i < 32; i++) {
            if ((n & (1 << i)) != 0) {
                if (last != -1) {
                    max = Math.max(max, i - last);
                }
                last = i;
            }
        }
        return max;
        
    }
}